platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronisedInstallerImpl.java

```diff
@@ -0,0 +1,566 @@
+    Callable<Boolean> isBootFinishedAvailable =
+        () -> {
+          LOGGER.debug("Waiting for BootFinished service to appear.");
+          return bundleContext.getServiceReference(BootFinished.class) != null;
+        };
+
+    wait(
+        isBootFinishedAvailable,
+        maxWaitTime,
+        DEFAULT_POLLING_INTERVAL,
+        String.format("BootFinished service did not appear within [%s] ms", maxWaitTime));
+    waitForBundles();
+  }
+
+  @Override
+  public Configuration createManagedFactoryService(
+      String factoryPid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    return createManagedFactoryService(
+        DEFAULT_MAX_SERVICE_WAIT, factoryPid, properties, bundleLocation);
+  }
+
+  @Override
+  public Configuration createManagedFactoryService(
+      long maxWaitTime, String factoryPid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    LOGGER.info("Creating managed service of factorypid [{}]", factoryPid);
+
+    Configuration createdConfig;
+    try {
+      createdConfig = configAdmin.createFactoryConfiguration(factoryPid, null);
+      createdConfig.setBundleLocation(bundleLocation);
+    } catch (IOException e) {
+      throw new SynchronisedInstallerException(
+          "Failed to initialize managed service factory configuration with fpid of [" + factoryPid + "]",
+          e);
+    }
+
+    try {
+      if (properties == null || properties.isEmpty()) {
+        createdConfig.update();
+      } else {
+        createdConfig.update(toDic(properties));
+      }
+    } catch (IOException e) {
+      throw new SynchronisedInstallerException(
+          "Failed to update created managed service factory configuration with pid of ["
+              + createdConfig.getPid()
+              + "]",
+          e);
+    }
+
+    waitForServiceAvailability(maxWaitTime, createdConfig.getPid());
+    return createdConfig;
+  }
+
+  @Override
+  public void updateManagedService(
+      String servicePid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    updateManagedService(DEFAULT_MAX_SERVICE_WAIT, servicePid, properties, bundleLocation);
+  }
+
+  @Override
+  public void updateManagedService(
+      long maxWaitTime, String servicePid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    waitForServiceAvailability(maxWaitTime, servicePid);
+
+    LOGGER.debug(
+        "Registering service configuration listener to listen to service with pid of {}.",
+        servicePid);
+    ServiceConfigurationListener listener = new ServiceConfigurationListener(servicePid);
+    ServiceRegistration<?> registration =
+        bundleContext.registerService(ConfigurationListener.class.getName(), listener, null);
+
+    Callable<Boolean> isServiceUpdated =
+        () -> {
+          LOGGER.debug(
```

Comment:
✏️ This trace will appear in every loop where we check, we might want to move it to `trace` or simply remove it since in any case, we are not really waiting here but just checking if the listener was called or not.